extmod/btstack/modbluetooth_btstack.c

```diff
@@ -612,14 +769,44 @@ int mp_bluetooth_gatts_notify(uint16_t conn_handle, uint16_t value_handle) {
 
 int mp_bluetooth_gatts_notify_send(uint16_t conn_handle, uint16_t value_handle, const uint8_t *value, size_t *value_len) {
     DEBUG_EVENT_printf("mp_bluetooth_gatts_notify_send\n");
-    // TODO: We need to use att_server_request_to_send_notification here as the stack may not be ready to send a notification.
+
+    // Attempt to send immediately, will copy buffer.
+    MICROPY_PY_BLUETOOTH_ENTER
     int err = att_server_notify(conn_handle, value_handle, value, *value_len);
-    return btstack_error_to_errno(err);
+    MICROPY_PY_BLUETOOTH_EXIT
+
+    if (err == BTSTACK_ACL_BUFFERS_FULL) {
+        DEBUG_EVENT_printf("mp_bluetooth_gatts_notify_send: ACL buffer full, scheduling callback\n");
+        // Schedule callback, making a copy of the buffer.
+        mp_btstack_pending_op_t *pending_op = btstack_enqueue_pending_operation(MP_BLUETOOTH_BTSTACK_PENDING_NOTIFY, conn_handle, value_handle, value, *value_len);
+
+        att_server_request_to_send_notification(&pending_op->context_registration, conn_handle);
+        // We don't know how many bytes will be sent.
+        *value_len = 0;
+
+        return 0;
+    } else {
+        return btstack_error_to_errno(err);
+    }
 }
 
 int mp_bluetooth_gatts_indicate(uint16_t conn_handle, uint16_t value_handle) {
```

Comment:
I just noticed: it doesn't look like this function is ever used??

Reply:
I've added `gatts_indicate(conn_handle, value_handle)` to modbluetooth. (And added to the ble_characteristic.py multitest)